packages/vouching/contracts/Vouching.sol

```diff
@@ -0,0 +1,101 @@
+pragma solidity ^0.4.24;
+
+
+import "./ZepToken.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+contract Vouching {
+  using SafeMath for uint256;
+
+  struct Dependency {
+    address owner;
+    address dependencyAddress;
+    uint256 stake;
+  }
+
+  mapping (string => Dependency) private _registry;
+  uint256 private _minimumStake;
+  ZepToken private _token;
+
+  modifier onlyDependencyOwner(string name) {
+    require(msg.sender == _registry[name].owner);
+    _;
+  }
+
+  event OwnershipTransferred(address indexed oldOwner, address indexed newOwner);
+  event DependencyCreated(string name, address indexed owner, address indexed dependencyAddress, uint256 initialStake);
+  event Vouched(string name, uint256 amount);
+  event Unvouched(string name, uint256 amount);
+  event DependencyRemoved(string name);
```

Comment:
I'd consider indexing the `name` argument in all events. The only issue would be losing track of the original name in `DependencyCreated`, but we could add a duplicated `not-indexed-name` argument there. I'd like to know what the OZ team thinks about this, though (ping @frangio, @elopio, @nventuro).

Reply:
Following a similar discussion: should `DependencyCreated` include an `initialStake`, or should we fire two subsequent events of `DependencyCreated+Vouched`? I think I prefer the current approach though.

Reply:
> 
> 
> Following a similar discussion: should `DependencyCreated` include an `initialStake`, or should we fire two subsequent events of `DependencyCreated+Vouched`? I think I prefer the current approach though.


We're *creating* an entry in the registry, so firing an event that is called `Vouched` in addition to the creation event might be somewhat confusing. IMO a single event (`DependencyCreated`) that includes the `initialStake` is clearer.

Reply:
I'm not sure if indexing strings is still being not simple, here is some research I did for the previous release (see https://github.com/zeppelinos/zos-lib/issues/199)

Reply:
I remembered there was something about indexing strings, that's why I didn't.. let's wait for the OZ guys input on this!